Makefile

```diff
@@ -1,14 +1,26 @@
+ifndef DEADFISH_DOCKER_NOSUDO
+dockercmd=sudo docker
+else
+dockercmd=docker
+endif
+
 all: server client-wasm client-native
-.PHONY: server client-wasm client-native test
+.PHONY: server client-wasm client-native test recmake-dfclient-wasm recmake-dfclient-native
 
 server:
-	sudo docker exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/server/build && make -j5'
+	$(dockercmd) exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/server/build && make -j5'
 
 client-wasm:
-	sudo docker exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/client/dfclient-wasm && make -j5'
+	$(dockercmd) exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/client/dfclient-wasm && make -j5'
 
 client-native:
-	sudo docker exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/client/dfclient-native && make -j5'
+	$(dockercmd) exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/client/dfclient-native && make -j5'
+
+recmake-dfclient-wasm:
+	$(dockercmd) exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/client/ && ./recmake_dfclient_wasm.sh'
+
+recmake-dfclient-native:
+	$(dockercmd) exec --user ubuntu -it deadfish-env /bin/bash -c 'cd /deadfish/client/ && ./recmake_dfclient_native.sh'
```

Comment:
I'll add support for sudo-less usage of docker in `dfctl`, too.

As for `dfctl` vs. `make`, that's probably up to @capturcus to decide. Personally, I prefer `make` because I don't have to install anything (with sudo, of course!), and I stil get autocompletion.

Reply:
@piodul 's reasoning about `make` makes sens to me. Because of using a different shell I also don't get autocompletion with `dfctl`. We should probably migrate everything to make, but that's a task for a different PR.